src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/FindSmallIndelRegions.java

```diff
@@ -114,49 +253,140 @@ public Finder( final ReadMetadata readMetadata, final SVReadFilter filter ) {
+                advanceBlock(sameContig, read);
             }
             final IntHistogram[] histoPair =
                     libraryToHistoPairMap.get(readMetadata.getLibraryName(read.getReadGroup()));
             histoPair[fillIdx].addObservation(Math.abs(read.getFragmentLength()));
         }
 
         public void checkHistograms( final List<BreakpointEvidence> evidenceList ) {
+            if ( curContig == null ) return;
+
             final int oldIdx = fillIdx ^ 1; // bit magic sets oldIdx to 1 if fillIdx is 0, and vice versa
-            SVInterval curInterval = null;
-            for ( final Map.Entry<String,IntHistogram[]> entry : libraryToHistoPairMap.entrySet() ) {
+            final int start = Math.max(1, curEnd - 2 * BLOCK_SIZE);
+            final SVInterval curInterval = new SVInterval(readMetadata.getContigID(curContig), start, curEnd);
+            for ( final Map.Entry<String, IntHistogram[]> entry : libraryToHistoPairMap.entrySet() ) {
                 final IntHistogram[] histoPair = entry.getValue();
                 final IntHistogram oldHisto = histoPair[oldIdx];
                 oldHisto.addObservations(histoPair[fillIdx]);
                 final long readCount = oldHisto.getTotalObservations();
                 if ( readCount > 0 &&
-                        readMetadata.getLibraryStatistics(entry.getKey())
-                                .isDifferentByKSStatistic(oldHisto,KS_SIGNIFICANCE) ) {
-                    if ( curInterval == null ) {
-                        int start = curEnd - 2 * BLOCK_SIZE;
-                        if ( start < 1 ) start = 1;
-                        curInterval = new SVInterval(readMetadata.getContigID(curContig), start, curEnd);
-                    }
+                        readMetadata.getLibraryStatistics(entry.getKey()).getCDF()
+                                .isDifferentByKSStatistic(oldHisto, KS_SIGNIFICANCE) ) {
                     evidenceList.add(
-                            new BreakpointEvidence.TemplateSizeAnomaly(curInterval,EVIDENCE_WEIGHT,(int)readCount));
+                        new BreakpointEvidence.TemplateSizeAnomaly(curInterval, EVIDENCE_WEIGHT, (int)readCount));
+                }
+                oldHisto.clear();
+            }
+        }
+
+        public void testReadAndGatherStatus( final GATKRead read, final List<String> statusList ) {
+            if ( !filter.isTemplateLenTestable(read) ) return;
+
+            final boolean sameContig;
+            if ( !(sameContig = read.getContig().equals(curContig)) || read.getStart() >= curEnd ) {
+                getStatus(statusList);
+                advanceBlock(sameContig, read);
+            }
+            final IntHistogram[] histoPair =
+                    libraryToHistoPairMap.get(readMetadata.getLibraryName(read.getReadGroup()));
+            histoPair[fillIdx].addObservation(Math.abs(read.getFragmentLength()));
+        }
+
+        public void getStatus( final List<String> statusList ) {
```

Comment:
Not sure what "status" means here, so adding a documentation (if worth it) would be helpful.